File: cap.10_linguagem_c++/codigos/exercicios/DIAGONAL_NEGATIVOS.cpp
# include <iostream>

using namespace std;

int main() {
    int N, qtdNegativos = 0;

    cout << "Qual a ordem da matriz? ";
    cin >> N;

    int arr[N][N];

    for (int i = 0; i < N; i++) {
        for (int j = 0; j < N; j++) {
            cout << "Elemento [" << i << "," << j << "]: ";
            cin >> arr[i][j];
        }
    }

    cout << "DIAGONAL PRINCIPAL:" << endl;
    for (int i = 0; i < N; i++) {
        cout << arr[i][i] << " ";
    }

    cout << endl;

    for (int i = 0; i < N; i++) {
        for (int j = 0; j < N; j++) {
            if (arr[i][j] < 0) {
                qtdNegativos++;
            }
        }
    }

    cout << "QUANTIDADE DE NEGATIVOS = " << qtdNegativos;

    return 0;
}

File: cap.11_java/codigos/exercicios/src/DiagonalNegativos.java
import java.util.Locale;
import java.util.Scanner;

public class DiagonalNegativos {

    public static void main(String[] args) {
        
        Locale.setDefault(Locale.US);
        Scanner sc = new Scanner(System.in);

        int N, qtdNegativos = 0;

        System.out.print("Qual a ordem da matriz? ");
        N = sc.nextInt();

        int[][] arr = new int[N][N];

        for (int i = 0; i < N; i++) {
            for (int j = 0; j < N; j++) {
                System.out.print("Elemento [" + i + "," + j + "]: ");
                arr[i][j] = sc.nextInt();
            }
        }

        System.out.println("Diagonal principal:");
        for (int i = 0; i < N; i++) {
            System.out.print(arr[i][i] + " ");
        }

        for (int i = 0; i < N; i++) {
            for (int j = 0; j < N; j++) {
                if (arr[i][j] < 0) {
                    qtdNegativos++;
                }
            }
        }
        System.out.println();
        System.out.println("QUANTIDADE DE NEGATIVOS = " + qtdNegativos);

        sc.close();
    }

}
